### PR TITLE
Remove qio_glocale_utf8 and qio_set_glocale()

### LINKED ISSUE #19193: Can we stop requiring particular C LOCALE / LANG settings?
There is a requirement on the environment that Chapel programs are run in that is buried in our documentation:

https://chapel-lang.org/docs/usingchapel/chplenv.html#character-set

> Chapel works with the Unicode character set with UTF-8 encoding and the traditional C collating sequence. Users are responsible for making sure that they are running Chapel in a suitable environment. For example, for en_US locale, the following environment variables should be set:
> 
>     LANG=en_US.UTF-8
>     LC_COLLATE=C
>     LC_ALL=""
> 
> Note: Other character sets may be supported in the future.

The situation is that, in practice, the I/O code currently assumes that non-binary data is stored as UTF-8. This is connected to the fact that our string type always stores UTF-8 and we don’t have any character set conversion in the standard library (see also #12726).

However, there is some old code in the I/O library that reads a "character" according to the C definition of a multibyte character. It does this if `qio_glocale_utf8` is not set to UTF-8 or ASCII. But, this will lead to garbles / bad encoding errors because storing it in to a string will assume it is a UTF-8 character. But besides that, we have seen some bugs with it that only impact people with a different C LANG setting.

The code that sets up qio_glocale_utf8 is here:

https://github.com/chapel-lang/chapel/blob/6702b173cfc8fffe10185e7f145c2c2e2ba55b33/runtime/src/chpl-init.c#L176-L180
 
https://github.com/chapel-lang/chapel/blob/6702b173cfc8fffe10185e7f145c2c2e2ba55b33/runtime/src/qio/qio_formatted.c#L36-L57

In terms of design in this area, it’s my view that we need to pick one of these two options:
a. all textual I/O is done in UTF-8 and if that’s not what you want, do binary I/O and then use a library to convert. (AFAIK this is what Rust does)
b. all textual I/O is done with a configurable character set that can be specified when opening the file/channel/reader/writer and is then converted to Unicode / UTF-8 during I/O (because strings are always UTF-8). (AFAIK this is what Python, Java, and C# do).

Note that neither of these depend on the C LOCALE / LANG environment variables. Also, we could arguably do (a) now and later add (b) as a non-breaking change. So maybe we could completely remove `qio_glocale_utf8` and switch to assuming textual data in I/O is UTF-8 in all cases in the near term. (The reason I say "maybe" is that there might be some C code in the library somewhere that uses the C multibyte infrastructure even for UTF-8. I don't think there is, but it is something to check.)

### PR DESCRIPTION
**Pull request by mppf**
Instead of detecting a UTF-8 locale, make sure to tell the C library to support UTF-8
during program initialization.

Resolves #19193.

This PR adds a comment to start_test but doesn't take out the setting of C LOCALE to en_US.UTF-8. That is because it could impact things other than chapel code (prediff scripts, etc). See #10213 #10302 for some history in this area.

- [x] full local testing
- [x] test/io/ferguson/utf8/widecols.chpl has the same output with or without `LANG=C` in the environment (this was not the case before)
- [x] hello world builds and runs on Mac OS Monterey
- [x] hello world builds and runs on Ubuntu 21.10.
- [x] `make check` OK on many vagrant vms (listed below)

```
bento-amazonlinux-2	fedora-32-cloud-base	     opensuse-Leap-15.3.x86_64
centos-7-devtoolset-11	fedora-33-cloud-base	     ubuntu-bionic64
centos-stream8		fedora-34-cloud-base	     ubuntu-focal64
debian-bullseye64	fedora-35-cloud-base	     ubuntu-impish64
debian-buster64		freebsd-FreeBSD-12.2-STABLE  ubuntu-impish64-homebrew
debian-stretch64	freebsd-FreeBSD-12.3-STABLE  ubuntu-xenial32
fedora-31-cloud-base	generic-alpine314	     ubuntu-xenial64
```

Reviewed by @jhh67 - thanks!

### CODE REVIEW

**Comment by ronawho** [comment]
FYI we have launcher code that tries to propagate `LANG`/`LC_ALL`/`LC_COLLATE`. I don't really understand any of the LOCALE/LANG stuff, but I was wondering if we can get rid of that code now. See `chpl_get_charset_env_args`, which is used for most launchers and the charset env var handing in `propagate_environment` in the `{slurm,pbs}-gasnetrun_ibv` ones.

File: runtime/include/qio/qio_formatted.h [hunk 1]
```diff
@@ -46,12 +46,10 @@
 extern "C" {
 #endif
 
-extern int qio_glocale_utf8; // for testing use.
-#define QIO_GLOCALE_UTF8 1
-#define QIO_GLOCALE_ASCII 2
-#define QIO_GLOCALE_OTHER -1
-
-void qio_set_glocale(void);
+//extern int qio_glocale_utf8; // for testing use.
```
Comment (gbtitus) [inline runtime/include/qio/qio_formatted.h:49]: Why not just remove these? The SCCS system can resurrect it if we ever need it. In my experience this sort of thing never gets cleaned up and we just end up with a bunch of commented-out code with nobody knowing why it's there.

Comment (mppf) [reply runtime/include/qio/qio_formatted.h:49]: Yes I meant to delete this comment before we got to the PR review but forgot. So thanks for the reminder!

**Review by jhh67 — APPROVED** [review]
👍

**Pull request merged**